src/pages/guides/tutorials/grids-addon.md

```diff
@@ -0,0 +1,1321 @@
+  const deleteGridBtn = document.getElementById("deleteGrid");
+
+  deleteGridBtn.onclick = async (event) => {
+    await scriptApi.deleteGrid();
+  };
+
+  createGridBtn.onclick = async (event) => {
+    await scriptApi.addGrid({
+      columns: colsInput.value,
+      rows: rowsInput.value,
+      gutter: gutterInput.value,
+      columnColor: colsColorPicker.value,
+      rowColor: rowsColorPicker.value,
+    });
+  };
+
+  // Only now we can enable the button
+  createGridBtn.disabled = false;
+  deleteGridBtn.disabled = false;
+});
+```
+
+Eventually, the two CTA buttons (Delete and Create) invoke methods exposed by the Document API, respectively `deleteGrid()` and `createGrid()`. The latter expects an options object with `rows`, `columns`, `gutter`, `columnColor`,  and `rowColor` properties.
+
+### Creating Rows and Columns
+
+It makes sense to approach this grid business with some caution, as we're just starting with the Document API. Let's set up `script/code.js` to expose this `addGrid()` method and manage the argument provided.
+
+<!-- Code below -->
+<CodeBlock slots="heading, code" repeat="1" languages="script/code.js"/>
+
+#### script/code.js
+
+```js
+import addOnScriptSdk from "AddOnScriptSdk";
+const { runtime } = addOnScriptSdk.instance;
+import { editor, utils, Constants } from "express";
+
+function start() {
+  runtime.exposeApi({
+    addGrid({ columns, rows, gutter, columnColor, rowColor }) {
+      console.log("addGrid", columns, rows, gutter, columnColor, rowColor);
+    },
+  });
+}
+
+start();
+```
+
+When the user clicks the Create button, the parameters from the UI are properly collected, passed to `addGrid()` in the Authoring Sandbox, and logged. So far, so good, the Communication API does its job.
+
+![](images/grids-addon-console.png)
+
+To begin with, we'll create rows: rectangles that must be as wide as the page. To calculate their height, first, subtract the total number of gutters (which is equal to the number of rows plus one) from the page height. Then, divide the resulting value by the number of rows.
+
+```text
+rowHeight = (pageHeight - (rowsNumber + 1) * gutter) / rowsNumber;
+```
+
+This is because we're using gutters as page margins, too, as the following illustration shows.
+
+![](images/grids-addon-rowheight.png)
+
+We must get hold of the [Document](/references/scriptruntime/editor/classes/Editor.md#documentroot) (as `documentRoot`, from the Editor class) and [Page](/references/scriptruntime/editor/classes/PageNode/)—the first one from the `pages` list will be OK for our purposes. Page properties like `width` and `height` will be used to compute the attributes of each "row" Rectangle.
+
+```js
+// ...
+runtime.exposeApi({
+	addGrid({ columns, rows, gutter, columnColor, rowColor }) {
+		const doc = editor.documentRoot;
+		const page = doc.pages.first;
+		const rowWidth = page.width;
+	  const rowHeight = (page.height - (rowsNumber + 1) * gutter) / rowsNumber;
+	},
+});
+```
```

Comment:
This _works_, but only for the first page. 

If we wanted to target the current page, we'd need to traverse up from the current insertion parent to determine the current page.

It might not be a bad idea to show how to do this, since we don't want to encourage the pattern of add-ons only working with the first page of the current document.

(I'm wondering now if we should also expose a `currentPage` property somewhere in the API to make this even easier.)

Reply:
Added the snippet showing it.
And yes, please, `currentPage` would be a nice addition :-)